vars/exec.groovy

```diff
@@ -33,3 +33,38 @@ def outputOf(arglist) {
 def statusOf(arglist) {
    return sh(script: shellEscapeList(arglist), returnStatus: true);
 }
+
+// Result of executing a shell command.
+class ExecResult {
+   // Escaped command that was run.
+   String command
+
+   // Resulting exit code.
+   Integer exitCode
+
+   // Combined output of stdout and stderr.
+   String output
+}
+
+// Execute a list of args as a shell command, then return the exit code, the
+// combined stdout + stderr, and the escaped command that was run.
+ExecResult runCommand(List<String> arglist) {
+   String cmd = shellEscapeList(arglist)
+   // Redirecting cmd stderr to stdout, in order to return it if cmd failed
+   def output = ""
+   def rc = 0
+   withEnv(["RC=0"]) {
+      output = sh(script: """
+         set +x
+         (${cmd}) 2>&1 || RC=\$?
```

Comment:
```suggestion
         ( ${cmd} ) 2>&1 || RC=\$?
```

This may not matter for the shells we use, but some shells need the whitespace.

Reply:
FWIW, I think I was totally wrong here -- all shells are ok without spaces.  (Of course, it's totally fine to put them in anyway.)  I think I was thinking of `[`, some shells definitely have problems with `[-n "$string"]`.